src/app/components/category-list.tsx

```diff
@@ -0,0 +1,74 @@
+"use client";
+
+import {useEffect, useState} from "react";
+import {Box, Card, CircularProgress, IconButton} from "@mui/material";
+import styles from "./category-list.module.css";
+import {Delete, Edit} from "@mui/icons-material";
+
+export function CategoryList() {
+    const [loading, setLoading] = useState<boolean>(false);
+    const [categories, setCategories] = useState<Category[]>([]);
+
+    useEffect(() => {
+        fetchCategories();
+    }, [])
+
+    const fetchCategories = async () => {
+        setLoading(true);
+        fetch(`/api/category`)
+            .then((res) => {
+                if (!res.ok) {
+                    throw new Error("Error fetching categories");
+                }
+
+                return res.json();
+            })
+            .then((data: { data: Category[] }) => {
+                setLoading(false);
+                setCategories(data.data);
+            });
+    }
+
+    const handleDelete = (id: number) => {
+        fetch(`/api/category/${id}`, {
+            method: "DELETE",
+            headers: {
+                "Content-Type": "application/json",
+            }
+        }).then(res => res.text())
+            .then((message: string | { success: boolean }) => {
+                if (typeof message === "string") {
+                    throw new Error(message);
+                }
+
+                fetchCategories();
+            })
+    }
```

Comment:
Kategorien können nicht gelöscht werden, wenn sie noch in der image-Tabelle referenziert werden.
Das macht zwar so Sinn, aber dem User wird das nicht angezeigt. 
Vllt sollte man hier vorher einmal separat testen, ob die Kategorie noch verwendet wird, und falls ja, dem User eine Fehlermeldung geben.
Und ansonsten die DELETE request ausführen.

Reply:
Fehler des Backends wird jetzt im Frontend angezeigt